hex_blueprints/check_run_status.py

```diff
@@ -0,0 +1,128 @@
+@dataclass
+class HexResponse:
+    """
+    This is a utility class to maintain the http status code with the json response
+    """
+    status_code : int
+    response_json : dict
+    
+def get_args():    
+    """
+    Creates the argument parser for the CLI
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project-id",dest='project_id',required=True)
+    parser.add_argument('--api-token',dest = 'api_token',required=True)
+    parser.add_argument('--run-id',dest = 'run_id',required=False)
+    ## add a flag for input params 
+    args = parser.parse_args()
+    return args
+
+def get_run_status(project_id, api_token, run_id):
+    """
+    Returns the json with metadata of the last run of the project
+    """
+    pattern = re.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$") ## copied from Hex's error message
+    matched = pattern.match(project_id)
+    if not matched:
+        print(f"Project Id {project_id} is in the incorrect format. The expected format is {pattern.pattern}. Please copy the correct value from Hex")
+        sys.exit(ec.EXIT_CODE_INVALID_PROJECT_ID)
+    url = f'https://app.hex.tech/api/v1/project/{project_id}/run/{run_id}'
+    headers = {"Authorization" : f"Bearer {api_token}"}
+    try:
+        response = requests.get(url= url, headers = headers)
+        status_code = response.status_code
+        response_json = response.json()
+        ## go through the known response codes documented by the HEX api: https://learn.hex.tech/docs/develop-logic/hex-api/overview#404-not-found
+        if status_code == 404:
+            print("Request was not found. Please ensure that you have the proper project id and run id from Hex")
+            sys.exit(ec.EXIT_CODE_BAD_REQUEST)
+        elif status_code == 429:
+            print("The number of requests has exceeded the limit, HEX allows 60 requests per minute. Please wait 2 minutes for the limit to reset")
+            sys.exit(ec.EXIT_CODE_EXCESSIVE_REQUESTS)
+        elif status_code == 401:
+            print("Request was unable to be authenticated, please ensure that your API token is entered correctly and that it is not expired.")
+            sys.exit(ec.EXIT_CODE_AUTHENTICATION_ERROR)
+        elif status_code == 500:
+            print("There was a server side error when processing the request, please contact HEX Support at support@hex.tech for continued disturbances")
+            sys.exit(ec.EXIT_CODE_HEX_SERVER_ERROR)
+        elif status_code == 200:
+            return response_json
+        else:
+            print("Unknown error when processing request")
+            sys.exit(ec.EXIT_CODE_UNKNOWN_ERROR)
+
+    except Exception as e:
+        print(f"Could not authenticate with API token {api_token}. ProjectID {project_id} matched expected format. Please ensure that you have the correct API token and Project Id from Hex and that the api token is not expired.")
```

Comment:
Same note here to just make the message more generic, since we can't be sure which one is causing the problem.

Reply:
Same fix as above